packages/block-library/src/image/image.js

```diff
@@ -1043,10 +1047,45 @@ export default function Image( {
 		);
 	}
 
+	const setAttributeForFeatureImage = ( value ) => {
+		const isPostHasFeatureImage =
+			// eslint-disable-next-line @wordpress/data-no-store-string-literals
+			select( 'core/editor' ).getEditedPostAttribute( 'featured_media' );
+		if ( isPostHasFeatureImage ) {
+			// dispatch notice that post already has feature image
+			createErrorNotice( __( 'Post already has a featured image.' ), {
+				type: 'snackbar',
+			} );
+			return;
+		}
+
+		// set post featured image as current image.
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		dispatch( 'core/editor' ).editPost( { featured_media: attributes.id } );
+
+		setAttributes( { isFeatureImage: ! value } );
+	};
+
+	const makeFeatureImageFormatControl = (
+		<BlockControls>
+			<ToolbarGroup>
+				<ToolbarButton
+					icon="format-image"
+					label={ __( 'Make Feature Image' ) }
+					onClick={ () =>
+						setAttributeForFeatureImage( isFeatureImage )
+					}
+					isPressed={ isFeatureImage }
+				/>
```

Comment:
@paaljoachim @jameskoster  @scruffian 

I want to confirm again here: does the "Make Feature Image" button mean to _label_ the block to be used as a featured image? Or is it meant to _update_ the featured image with the current image?

For example, let's assume that the former (labeling) is what is expected here. Furthermore, suppose that [useFirstImageFromPost](https://github.com/WordPress/gutenberg/pull/56573) is enabled for the featured image and a second Image block in the content is labeled to be used as a featured image.

In this case, which one will actually be displayed as the featured image?

To prevent such inconsistencies, I prefer the latter approach (updating the featured image with the current image).

Reply:
The latter was my expectation. But I don't have a strong feeling about this feature and would welcome more feedback.

Reply:
I expect to use the image as a selected featured image.

That is when I press the link/button "Set featured image" associated with the Image block a featured image should also be added here: 
![Screenshot 2024-10-08 at 20 34 13](https://github.com/user-attachments/assets/668b079a-75fb-409d-83bf-8a447b1b16e5)

It becomes an additional option to directly set the Featured image through the ellipse (3 button drop down) in an Image block. As it is added here it could easily also be added to a Cover block, Media & Text block etc.
If there was already a featured image added to the post then clicking the option in the ellipse would then updated the featured image with the one that was just selected through the Image block.

Conclusion is that having an additional method to add a featured image would be helpful. 

--

Now on another similar issue.
It would also be helpful to set first image in post as featured image through Global settings. This means if one forgets to set a featured image then the first image in the post will automatically be set as featured. I have experienced over and over that customers forget to add a featured image. Having this happen automatically would very much help.